File: javascript/intermediate/longest-consecutive-sequence/longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    if (nums.length === 0) {
        return 0;
    }

    const numSet = new Set(nums);
    let longestStreak = 0;

    for (const num of nums) {
        // Check if the current number is the start of a sequence
        // (i.e., num - 1 is not in the set)
        if (!numSet.has(num - 1)) {
            let currentNum = num;
            let currentStreak = 1;

            // Count consecutive numbers
            while (numSet.has(currentNum + 1)) {
                currentNum++;
                currentStreak++;
            }

            longestStreak = Math.max(longestStreak, currentStreak);
        }
    }

    return longestStreak;
};

module.exports = longestConsecutive;

File: python/intermediate/longest-consecutive-sequence/longest_consecutive_sequence.py
"""
Solution for Longest Consecutive Sequence problem.

Given an unsorted array of integers nums, return the length of the longest 
consecutive elements sequence.

Time complexity: O(n)
Space complexity: O(n)
"""

from typing import List


def longest_consecutive(nums: List[int]) -> int:
    """
    Find the length of the longest consecutive elements sequence.
    
    Args:
        nums: List of integers
        
    Returns:
        Length of the longest consecutive sequence
        
    Example:
        >>> longest_consecutive([100,4,200,1,3,2])
        4
        >>> longest_consecutive([0,3,7,2,5,8,4,6,0,1])
        9
    """
    if not nums:
        return 0
    
    num_set = set(nums)
    longest_streak = 0
    
    for num in nums:
        # Check if the current number is the start of a sequence
        # (i.e., num - 1 is not in the set)
        if num - 1 not in num_set:
            current_num = num
            current_streak = 1
            
            # Count consecutive numbers
            while current_num + 1 in num_set:
                current_num += 1
                current_streak += 1
            
            longest_streak = max(longest_streak, current_streak)
    
    return longest_streak